lib/paper_trail.rb

```diff
@@ -62,32 +68,61 @@ def enabled?
       !!PaperTrail.config.enabled
     end
 
-    # Sets whether PaperTrail is enabled or disabled for the current request.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_controller=(value)
-      paper_trail_store[:request_enabled_for_controller] = value
+      ::ActiveSupport::Deprecation.warn(
+        "PaperTrail.enabled_for_controller= is deprecated, " \
+        "use PaperTrail.request.enabled_for_controller="
+      )
+      request.enabled_for_controller = value
     end
 
-    # Returns `true` if PaperTrail is enabled for the request, `false` otherwise.
-    #
-    # See `PaperTrail::Rails::Controller#paper_trail_enabled_for_controller`.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_controller?
-      !!paper_trail_store[:request_enabled_for_controller]
+      ::ActiveSupport::Deprecation.warn(
+        "PaperTrail.enabled_for_controller? is deprecated, " \
+        "use PaperTrail.request.enabled_for_controller?"
+      )
+      request.enabled_for_controller?
     end
 
-    # Sets whether PaperTrail is enabled or disabled for this model in the
-    # current request.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_model(model, value)
-      paper_trail_store[:"enabled_for_#{model}"] = value
+      ::ActiveSupport::Deprecation.warn(
+        "PaperTrail.enabled_for_model is deprecated, " \
+        "use PaperTrail.request.enabled_for_model"
+      )
+      request.enabled_for_model(model, value)
     end
 
-    # Returns `true` if PaperTrail is enabled for this model in the current
-    # request, `false` otherwise.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_model?(model)
-      !!paper_trail_store.fetch(:"enabled_for_#{model}", true)
+      ::ActiveSupport::Deprecation.warn(
+        "PaperTrail.enabled_for_model? is deprecated, " \
+        "use PaperTrail.request.enabled_for_model?"
+      )
+      request.enabled_for_model?(model)
+    end
+
+    # All methods that handle request-level configuration and are now handled
+    # by PaperTrail::Request.
+    # @api public
+    def request(options = nil)
+      if options.nil?
+        Request
+      else
+        begin
+          before = Request.to_h
+          Request.set(options)
```

Comment:
Do we want to limit what options are settable, and perhaps validate the options?  If they set:
```ruby
PaperTrail.request(some_nonexistant_option: true) do
 ...
end
```

What should happen?

What if they try to set a private attribute?  Such as `transaction_id`?